### ISSUE TITLE
Helm 3: "required" template function breaks linter

### ISSUE THREAD

**Issue by andytom** [helm/helm#5886]
Output of `helm version`:
```
$ helm version
version.BuildInfo{Version:"v3.0.0-alpha.1", GitCommit:"b9a54967f838723fe241172a6b94d18caf8bcdca", GitTreeState:"clean"}
```

Output of `kubectl version`:
N/A

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
N/A

If a chart uses the `required` filter with no default value running `helm lint` over the chart will fail, this does not happen in helm 2.

An example chart that will reproduce this issue is as follows:

**Chart.yaml**
```yaml
apiVersion: v1
name: test
description: A Helm chart for Kubernetes

# A chart can be either an 'application' or a 'library' chart.
#
# Application charts are a collection of templates that can be packaged into versioned archives
# to be deployed.
#
# Library charts provide useful utilities or functions for the chart developer. They're included as
# a dependency of application charts to inject those utilities and functions into the rendering
# pipeline. Library charts do not define any templates and therefore cannot be deployed.
type: application

# This is the chart version. This version number should be incremented each time you make changes
# to the chart and its templates, including the app version.
version: 0.1.0

# This is the version number of the application being deployed. This version number should be
# incremented each time you make changes to the application.
appVersion: 1.16.0
```
**values.yaml**
```yaml
---
``` 
**templates/foo.yaml**
```yaml
---
foo: {{ .Values.unset | required "uset is required!" }}
```

When running `helm lint` on this chart you get the following error:
```
Error: 1 chart(s) linted, 1 chart(s) failed
	%!s(<nil>)
	[INFO] Chart.yaml: icon is recommended
	[ERROR] templates/: render error at (test/templates/foo.yaml:3): 'uset is required!'
```

When running the same command using helm 2 the chart is valid
```
$ helm2 version --client
Client: &version.Version{SemVer:"v2.14.1", GitCommit:"5270352a09c7e8b6e8c9593002a73535276507c0", GitTreeState:"clean"}

$ helm2 lint .
2019/06/13 16:00:11 [INFO] Missing required value: uset is required!
==> Linting .
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, no failures
```

I believe this maybe a regression of #4747 and #4221.

**Comment by bacongobbler** [comment]
The patch was never brought over to Helm 3 prior to the removal of Tiller. Should be as simple as porting that patch over to the dev-v3 branch. Feel like taking a crack at that?

**Comment by andytom** [comment]
> The patch was never brought over to Helm 3 prior to the removal of Tiller. Should be as simple as porting that patch over to the dev-v3 branch. Feel like taking a crack at that?

I've opened a PR, the dev-v3 and master branch have drifted a bit but I think I've got everything across.

**Comment by andytom** [comment]
Fixed by #5889.
